src/server/appserver.cpp

```diff
@@ -333,14 +334,40 @@ AppServer::AppServer(int &argc, char **argv)
         long timestamp =
             std::chrono::time_point_cast<std::chrono::seconds>(std::chrono::system_clock::now()).time_since_epoch().count();
         std::string timestampStr = std::to_string(timestamp);
-        KDC::ParmsDb::instance()->updateAppState(AppStateKey::LastServerSelfRestart, timestampStr, found);
-        if (!KDC::ParmsDb::instance()->updateAppState(AppStateKey::LastServerSelfRestart, timestampStr, found) || !found) {
+        KDC::ParmsDb::instance()->updateAppState(AppStateKey::LastServerSelfRestartDate, timestampStr, found);
+        if (!KDC::ParmsDb::instance()->updateAppState(AppStateKey::LastServerSelfRestartDate, timestampStr, found) || !found) {
             LOG_WARN(_logger, "Error in ParmsDb::updateAppState");
             addError(Error(ERRID, ExitCodeDbError, ExitCauseDbEntryNotFound));
             throw std::runtime_error("Failed to update last server self restart.");
         }
     }
 
+    // Check if a log Upload has been interrupted
+    std::string logUploadStatusStr = "";
+    if (bool found = false;
+        !ParmsDb::instance()->selectAppState(AppStateKey::LogUploadStatus, logUploadStatusStr, found) || !found) {
+        LOG_WARN(_logger, "Error in ParmsDb::selectAppState");
+        addError(Error(ERRID, ExitCodeDbError, ExitCauseDbEntryNotFound));
+        throw std::runtime_error("Failed to get log upload status.");
+    }
+
+    char logUploadStatus = logUploadStatusStr.size() > 0 ? logUploadStatusStr[0] : 'N';
+    int logUploadProgress = logUploadStatusStr.size() > 1 ? std::stoi(logUploadStatusStr.substr(1)) : 0;
+    if (logUploadStatus == 'A' || logUploadStatus == 'U') {
+        LOG_WARN(_logger, "App was closed during log upload, aborting upload");
```

Comment:
At this moment, the upload is already aborted.
The log level should be Debug and not Warning (normal case).
A better log message could be "App was closed during log upload, resetting upload status".

Reply:
I applied your suggestion in commit: https://github.com/Infomaniak/desktop-kDrive/pull/117/commits/4f90b7794331b0621ebf2288f3810dd2ec5987ab